### PR TITLE
[Project] Delete serving config maps with project

### PR DESCRIPTION
**Pull request by alonmr**
https://iguazio.atlassian.net/browse/ML-6994

I noticed that on the system tests setup the serving functions config maps were created over a month ago so I checked why they were not deleted and found it was not implemented. Added the implementation here with some minor flavor changes.

### CODE REVIEW

File: server/api/api/utils.py [hunk 1]
```diff
@@ -1378,7 +1379,7 @@ async def delete_function(
                 return None
             except Exception as exc:
                 # return tuple with failure info
-                return function, str(exc)
```
Comment (liranbg) [inline server/api/api/utils.py:1381]: that might be intentional to avoid bloating the failure reason. @rokatyy correct?

**Review by liranbg — COMMENTED** [review]
minor

File: server/api/api/utils.py (hunk 1 above)
Comment (rokatyy) [reply server/api/api/utils.py:1381]: @liranbg yes, that's correct, I'd not change that

**Review by liranbg — APPROVED** [review]

File: server/api/api/utils.py (hunk 1 above)
Comment (alonmr) [reply server/api/api/utils.py:1381]: noted. though I would like to move `_delete_nuclio_functions_in_batches` to `server/api/crud/runtimes/nuclio/function.py`. Its a more fitting place IMO. wdyt?

Comment (rokatyy) [reply server/api/api/utils.py:1381]: @alonmr agree 👍

**Pull request merged**